lib/RestService/src/RestService.cpp

```diff
@@ -311,24 +346,65 @@ bool RestService::getResponse(void* restId, bool& isValidRsp, DynamicJsonDocumen
     bool isSuccessful = false;
     Msg  msg;
 
-    if (true == m_taskProxy.peek(msg))
+    if (true == m_isRunning)
     {
-        if (restId == msg.restId)
+        if (true == m_taskProxy.peek(msg))
         {
-            isValidRsp   = msg.isMsg;
-            payload      = msg.rsp;
-            isSuccessful = true;
-
-            if (false == m_taskProxy.receive(msg))
+            if (restId == msg.restId)
             {
-                LOG_FATAL("Two clients with the same restId exist!");
+                isValidRsp   = msg.isMsg;
+                payload      = msg.rsp;
+                isSuccessful = true;
+
+                if (false == m_taskProxy.receive(msg))
+                {
+                    LOG_FATAL("Two clients with the same restId exist!");
+                }
             }
         }
     }
+    else
+    {
+        isValidRsp = false;
+        payload    = nullptr;
+    }
 
     return isSuccessful;
 }
 
+void RestService::addToRemovedPluginIds(void* restId)
+{
+    if (nullptr == restId)
+    {
+        LOG_ERROR("Cannot add nullptr to removedPluginIds!");
+    }
+    else
+    {
+        removedPluginIds.push_back(restId);
+    }
+}
+
+void RestService::removeExpiredResponses()
+{
+    bool                 isValidRsp = false;
+    DynamicJsonDocument* jsonDoc    = nullptr;
+
+    for (size_t i = 0; i < (removedPluginIds.size()); i++)
```

Comment:
No single character variables please. And use pre-increment. And braces around the call is not required.